tools/local-network/local_network.py

```diff
@@ -100,12 +99,22 @@ def start_prometheus(self, log_branch, grafana_password, hosts):
         print(f'Starting prometheus, branch={log_branch}')
         os.mkdir(os.path.join(WORK_DIR, 'prometheus'))
         template = open(os.path.join(PROJECT_DIR, 'tools', 'local-network', 'prometheus.yml.template')).read()
+        # Add some stuff about grafana push if we have a grafana password
+        template_extra = '''
+remote_write:
+- url: https://prometheus-us-central1.grafana.net/api/prom/push
+  basic_auth:
+    username: 8687
+    password: ${GRAFANA_PASSWORD}
+        '''.replace('${GRAFANA_PASSWORD}', grafana_password) if grafana_password else ""
+
+        # Write promtheus.yml by replacing in template, and appending template_extra
         with open(os.path.join(WORK_DIR, 'prometheus.yml'), 'w') as f:
             f.write(template
                 .replace('${BRANCH}', log_branch)
-                .replace('${GRAFANA_PASSWORD}', grafana_password)
                 .replace('${HOSTS}', hosts)
```

Comment:
:thought_balloon: More of an FYI as this was here before, but python supports [Template strings]](https://docs.python.org/3/library/string.html#template-strings). The main benefit is that it will raise an exception if one fails to provide all of the substitutions

Reply:
I think the way Eran set this up with the files being external is nice because then you dont have to look at code to see what the config will be, i would prefer to avoid changing that